lib/lotus/model/adapters/sql/collection.rb

```diff
@@ -280,6 +283,14 @@ def _serialize(entity)
           def _deserialize(entity)
             @mapped_collection.deserialize([entity]).first
           end
+
+          def _apply_defaults(entity, defaults)
+            defaults.each_pair do |attribute, default|
+              if entity.send(attribute).nil?
```

Comment:
Maybe it's better to serialize the entity and apply the defaults to the hash and then deserialize